include/SH3/system/glbuffer.hpp

```diff
@@ -0,0 +1,120 @@
+/** @file
+ *
+ *  A wrapper and more OO approach to OpenGL Vertex Buffer Objects. This is also helps prevent any
+ *  memory problems as well as providing a whole lot of helper functions for our VBO.
+ *
+ *  @copyright 2017 Palm Studios
+ *
+ *  @date 8-6-2017
+ *
+ *  @author Jesse Buhagiar
+ */
+#ifndef buffer_HPP_INCLUDED
+#define buffer_HPP_INCLUDED
+
+#include "GL/glew.h"
+#include "GL/gl.h"
+
+#include <string>
+#include <cstddef>
+
+namespace sh3_gl
+{
+    /**
+     *
+     * An OpenGL Vertex Buffer Object (VBO). Provides a nice, easy, typesafe way
+     * to create and manage OpenGL buffers.
+     */
+    struct sh3_buffer final
+    {
+    public:
+
+        enum class Target
+        {
+            ARRAY_BUFFER    = GL_ARRAY_BUFFER,
+            ELEMENT_BUFFER  = GL_ELEMENT_ARRAY_BUFFER,
+            TEXTURE_BUFFER  = GL_TEXTURE_BUFFER
+        };
+
+        /**
+         *  Default constructor.
```

Comment:
This is part of #94, as it required some of its functionality. Is there a way I can fix this??